lib/atlas/initializer_input.rb

```diff
@@ -0,0 +1,23 @@
+module Atlas
+  class InitializerInput
+    include ActiveDocument
+
+    DIRECTORY = 'initializer_inputs'
+
+    attribute :query,           String
+    attribute :share_group,     Symbol
+    attribute :priority,        Integer, default: 0
+    attribute :unit,            String
+    attribute :user_value,      Float
```

Comment:
Yes and no. The other way is to convert the `init` Hash to a `Hash[InitializerInput => Float]` or something like that. The only problem that will give is that the initialization of the `init` needs  to be overwritten in the `DerivedDataset`.

Something like:

```ruby
# Pseudo code:
class DerivedDataset
  def initialize(attributes)
    super(attributes)
    set_init(attributes[:init])
  end
end
```

I'm not sure what has my preference. What is your idea about this?